addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -425,6 +490,28 @@ def _l10n_in_edi_generate_invoice_json(self, invoice):
                 json_payload["ExpDtls"].update({
                     "Port": invoice.l10n_in_shipping_port_code_id.code
                 })
+        if not invoice._l10n_in_edi_is_managing_invoice_negative_lines_allowed():
+            return json_payload
+
+        # Set nagative lines agains positive lines as discount with same HSN code
+        discount_lines = []
+        # reverse lines to remove from last
```

Comment:
Why the last first?  (not sure why you do that or what is the benefit as you need all discount lines anyways)  Would not we have problems with the enumerate when we delete lines? (might be a not important detail)

Reply:
```
>>> a = ['1','2','3','4','5']
>>> for i in a:
...     if i in ('3','4'):
...             a.remove(i)
... 
>>> a
['1', '2', '4', '5']
```
using enumerate is the same

```
>>> a = ['1','2','3','4','5']
>>> for index,i in enumerate(a):
...     if i in ('3','4'):
...             a.pop(index)
... 
>>> a
['1', '2', '4', '5']
>>> 

```
If we reverse it and then pop it then the index of next is not changed

```
>>> a = ['1','2','3','4','5']
>>> for i in a[::-1]:
...     if i in ('3','4'):
...             a.remove(i)
... 
>>> a
['1', '2', '5']

```

Reply:
@jva-odoo You could just do an a.copy() then instead of adding a lot of confusion then...